### PR TITLE
AWS: remove S3URI scheme restrictions

### PR DESCRIPTION
**Pull request by jackye1995**
As discussed in the mailing list: https://mail-archives.apache.org/mod_mbox/iceberg-dev/202112.mbox/%3CCAMwmD1_P2PCMpzUai5RV%2B4a9Bv62ZFqrrDdxCtLcTSvPnUxk6g%40mail.gmail.com%3E

Based on what Piotr said, 

> There are storages which are primarily used as "s3 compatible" and they need these settings to make them work.
We've seen these being used to access MinIO, Ceph and even S3 with some gateway

It seems to make more sense to just remove the restriction entirely instead of extending the list of allowed schemes.

@rdblue @danielcweeks @findepi @mayursrivastava

### CODE REVIEW

**Comment by jackye1995** [comment]
looks like flaky test, restart CI

File: aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java [hunk 1]
```diff
@@ -58,14 +54,14 @@
...
-    String scheme = schemeSplit[0];
-    ValidationException.check(VALID_SCHEMES.contains(scheme.toLowerCase()), "Invalid scheme: %s", scheme);
+    ValidationException.check(schemeSplit.length == 2,
+        "Cannot determine S3 URI scheme: %s", location);
 
     String [] authoritySplit = schemeSplit[1].split(PATH_DELIM, 2);
-    ValidationException.check(authoritySplit.length == 2, "Invalid S3 URI: %s", location);
-    ValidationException.check(!authoritySplit[1].trim().isEmpty(), "Invalid S3 key: %s", location);
+    ValidationException.check(authoritySplit.length == 2,
+        "Cannot determine bucket in S3 URI path: %s", location);
```
Comment (rdblue) [inline aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:62]: The bucket isn't in the URI path element, it is in the authority. Maybe just "Cannot determine bucket in S3 URI" instead?

File: aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java [hunk 2]
```diff
@@ -58,14 +54,14 @@
...
+    ValidationException.check(!authoritySplit[1].trim().isEmpty(),
+        "Cannot determine object key in S3 URI path: %s", location);
```
Comment (rdblue) [inline aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:64]: What about `Invalid S3 URI: path is empty`?

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks good overall. I'd probably fix some of the error messages that were updated to be more specific.

File: aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java (hunk 1 above)
Comment (jackye1995) [reply aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java]: I see. I always thought authority is a part of path... updated

**Comment by jackye1995** [comment]
Given Ryan approves this and all error messages are fixed, I am merging this. @danielcweeks Let me know if you have any additional concern.

**Pull request merged**